### PR TITLE
Added square particles CVar for GL 1.4

### PR DESCRIPTION
**Pull request by DenisBelmondo**
This is the proper, fixed version of #492. Thanks for alerting me as to the reason why the diffs were like that, still trying to find my way around git (or source control utils in general lol).

### CODE REVIEW

File: src/client/refresh/gl1/gl1_main.c [hunk 1]
```diff
@@ -1667,6 +1669,26 @@ RI_BeginFrame(float camera_separation)
...
+		switch ((int)gl1_particle_square->value)
+		{
+			default:
+				glDisable(GL_POINT_SMOOTH);
+				break;
+			case 0:
+				glEnable(GL_POINT_SMOOTH);
+				break;
+		}
+	}
```
Comment (Yamagi) [inline src/client/refresh/gl1/gl1_main.c]: This block must be wrapped with `if (gl_config.pointparameters)`. Without it GPUs that don't support the pointparameter extension may run into problem. At `gl1_pointparameter 0` might be broken.

**Comment by Yamagi** [comment]
Hi,
this is a good idea, but it doesn't work for me. No matter if I set `gl1_particle_square` to `1` or `0`, the particles are always rendered as dots and not squares. This is a Geforce 750TI with driver 440.31.

And if you're already on it... ;) In b7bf822e6d985c1fe1c88ae3355ea646f5561e4e we've changed the particles for the `gl1_pointparameter` case from squares to octagons. Maybe it would be a good idea to implement `gl1_particle_square` there as well?

File: src/client/refresh/gl1/gl1_main.c (hunk 1 above)
Comment (DanielGibson) [reply src/client/refresh/gl1/gl1_main.c]: also, why a switch for just two cases (instead of if/else)?

**Comment by DenisBelmondo** [comment]
Interesting, I'll definitely investigate today and also update my branch to include the if statement you mentioned earlier. Thanks for letting me know.

**Comment by DenisBelmondo** [comment]
Hi all, mistakes were made along the way, but I think I got it figured out. I am hoping that the cards that don't support point parameters will enjoy the little square pixel buffer I added right below the dot one :). Unfortunately with my current setup, I cannot verify if that fallback works, but I am very sure the logic checks out.

**Comment by Yamagi** [comment]
Hi,
thank you for the corrections. The normal case with `gl1_pointparameters 1` looks good and I'm fine with it. But there's still something wrong with `gl1_pointparameters 0` (requires `vid_restart`). With a very first sight on the code I don't see what could be the problem.

**Comment by DenisBelmondo** [comment]
Interesting. I'll see if I can do some more investigation soon. Thanks for letting me know.

**Comment by Yamagi** [comment]
I've merged this by hand. And added a commit that fixes the `gl1_pointparameters 0` case.

**Comment by DenisBelmondo** [comment]
Awesome, thanks! Got bogged down with work and holidays recently so I didn't get a chance to look at it very much. Glad to hear it's in now :)

**Pull request closed without merging**